src/Microsoft.AspNetCore.Razor.Evolution/Intermediate/BlockDirectiveIRNode.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Razor.Evolution.Legacy;
+
+namespace Microsoft.AspNetCore.Razor.Evolution.Intermediate
+{
+    public class BlockDirectiveIRNode : RazorIRNode
```

Comment:
How do you know which bits are the body and which bits are the token declaration? By fixing that you may be able to have a single directive node since RazorIRNode always has children available.